src/algorithms/models/graph.ts

```diff
@@ -5,43 +5,40 @@ import Vertex from './vertex';
  */
 
 export default class Graph {
-    /**
-     * Map objects for the graph sample
-     */
-    private readonly _vertices: Map<number, Vertex>;
+  /**
+   * Map objects for the graph sample
+   */
+  private readonly _vertices: Map<number, Vertex>;
 
-    /**
-     * Initializes new Map objects for the Vertices displayed on the graph
-     */
-    constructor() {
-        this._vertices = new Map<number, Vertex>();
-    }
+  /**
+   * Initializes new Map objects for the Vertices displayed on the graph
+   */
+  constructor() {
+    this._vertices = new Map<number, Vertex>();
+  }
 
-    /**
-     * Adds a Vertex object or number of Vertex objects into the vertices Map object
-     * @param x : Vertex object
-     */
-    addVertex(...x: Vertex[]): void {
-        for (const v of x) {
-            this._vertices.set(v.id, v);
-        }
+  /**
+   * Adds a Vertex object or number of Vertex objects into the vertices Map object
+   * @param x : Vertex object
+   */
+  addVertex(...x: Vertex[]): void {
+    for (const v of x) {
+      this._vertices.set(v.id, v);
     }
+  }
 
-    /**
-     * Get function which returns a certain Vertex inside the Vertices dictionary
-     * @param id : The ID of the Vertex object to be returned
-     */
-    getVertex(id: number): Vertex {
-        if (!this.vertices.has(id)) {
-            throw new Error('Vertex ID not found!');
-        }
-        return this.vertices.get(id);
-    }
+  /**
+   * Get function which returns a certain Vertex inside the Vertices dictionary
+   * @param id : The ID of the Vertex object to be returned
+   */
+  getVertex(id: number): Vertex {
+    return this._vertices.get(id);
+  }
```

Comment:
I understand that throwing an error might be too strict but the method should still check whether the vertex exists or not. What will be returned in that case? It should be at least documented.

Reply:
Still, I might like the error more. It should never happen and if it does, there is a bug.

Reply:
The method will return undefined, and we can check for that outside the method. Undefined is by definition a 'falsy' value (it evaluates to false in a boolean context like an if). Other 'falsy' values include null, 0, empty string and NaN.
e. g.
`let vertex = getVertex(someNodeId)
if(vertex) {
 // Vertex was found -> do stuff
}
else {
 // Vertex was not found -> do other stuff
}
`
Maybe we need to search for a node that wasn't in the graph initially. 
For example, we ship a few ways, but some nodes in them are missing in the graph. We could build-up a 'missingNodes' list and query for them specifically. A tie-in idea with this would be to query for ways first, and then query nodes as they are missing when building the graph.
An error would kill the process, though, unless encapsulated in a try {} catch, but that could create some complications. We can discuss this more, and revert it back, if you guys think it's necessary.